### ISSUE TITLE
Stage issue: floor rendered at the wrong height

### ISSUE THREAD

**Issue by K4thos** [ikemen-engine/Ikemen-GO#172]
Mugen:
![unknown](https://user-images.githubusercontent.com/8928195/92315589-f8455a80-efe7-11ea-9c32-5a6574eaee3f.png)

Ikemen GO:
![unknown2](https://user-images.githubusercontent.com/8928195/92315590-fe3b3b80-efe7-11ea-8f6e-ffa158f930f7.png)

Stage link: https://mega.nz/folder/309ySC5L#r7eVXVIK7KanbVw5DQnYYw/folder/vxNAHagI

**Comment by K4thos** [comment]
The issue with wrong height seems to be related to `zoomin` value set to < 1.

Stage being cropped at top is caused by this code: in system.go draw function:
```go
		if off < (yofs-y+s.cam.boundH)*scl {
			rect[3] = (int32(math.Ceil(float64(((yofs-y+s.cam.boundH)*scl-off)*
				float32(s.scrrect[3])))) + s.gameHeight - 1) / s.gameHeight
			fade(rect, 0, 255)
		}
		if off > (-y+yofs2)*scl {
			rect[3] = (int32(math.Ceil(float64(((y-yofs2)*scl+off)*
				float32(s.scrrect[3])))) + s.gameHeight - 1) / s.gameHeight
			rect[1] = s.scrrect[3] - rect[3]
			fade(rect, 0, 255)
		}
```
I wasn't able to fix it myself.

**Comment by K4thos** [comment]
another example of this problem:
[InfernoBonus-16.zip](https://github.com/Windblade-GR01/Ikemen-GO/files/6795800/InfernoBonus-16.zip)

ikemen:
![ikemen026](https://user-images.githubusercontent.com/8928195/125174134-7273dc00-e1c3-11eb-8f7e-d91af4e478f1.png)

mugen:
![mugen021](https://user-images.githubusercontent.com/8928195/125174128-67b94700-e1c3-11eb-916c-f0854a127dcc.png)

**Comment by K4thos** [comment]
one more:
[OneHellOfATime2.zip](https://github.com/Windblade-GR01/Ikemen-GO/files/7002662/OneHellOfATime2.zip)
ikemen:
![ikemen000](https://user-images.githubusercontent.com/8928195/129792665-40b624a4-bbbb-4ac1-9f23-70802525d161.png)
mugen 1.1:
![mugen048](https://user-images.githubusercontent.com/8928195/129792704-f93ab7b2-2164-4b77-8e4d-a36d04dbea6f.png)
again most likely related to zoomin < 1 (0.5)

**Comment by K4thos** [comment]
fixed by #605
